### ISSUE TITLE
Feature: Payments / SuperChats?

### ISSUE THREAD

**Issue by dm17** [GRVYDEV/Project-Lightspeed#6]
I see a lot of similar projects popping up and I'm still trying to brainstorm the best route to take for a SuperChats alternative for self-hosted streams. How can participants help the artist? OBS plugins? All ideas welcome :) 
Thanks!

**Comment by GRVYDEV** [comment]
So my inital thought would to be incorporating either Stripe or Paypal

**Comment by dm17** [comment]
> So my inital thought would to be incorporating either Stripe or Paypal

Sweet. I would stick with Stripe for now since the API is way better and so far a lot fewer anti-competitive & overall unfair practices.

How do you see that working? What would be the requirements?

**Comment by GRVYDEV** [comment]
> > So my inital thought would to be incorporating either Stripe or Paypal
> 
> Sweet. I would stick with Stripe for now since the API is way better and so far a lot fewer anti-competitive & overall unfair practices.
> 
> How do you see that working? What would be the requirements?

So we would need to add some sort of user registration/account system and then we would add the API flow for someone to subscribe to the stream and then the person hosting the stream server would configure their stripe account and set their API keys and it should work!

**Comment by dm17** [comment]
> > > So my inital thought would to be incorporating either Stripe or Paypal
> > 
> > 
> > Sweet. I would stick with Stripe for now since the API is way better and so far a lot fewer anti-competitive & overall unfair practices.
> > How do you see that working? What would be the requirements?
> 
> So we would need to add some sort of user registration/account system and then we would add the API flow for someone to subscribe to the stream and then the person hosting the stream server would configure their stripe account and set their API keys and it should work!

You've seen how SuperChats on YouTube work? It is pretty slick... It would integrate with the chatroom?

**Comment by GRVYDEV** [comment]
> > > > So my inital thought would to be incorporating either Stripe or Paypal
> > > 
> > > 
> > > Sweet. I would stick with Stripe for now since the API is way better and so far a lot fewer anti-competitive & overall unfair practices.
> > > How do you see that working? What would be the requirements?
> > 
> > 
> > So we would need to add some sort of user registration/account system and then we would add the API flow for someone to subscribe to the stream and then the person hosting the stream server would configure their stripe account and set their API keys and it should work!
> 
> You've seen how SuperChats on YouTube work? It is pretty slick... It would integrate with the chatroom?

I would have to dig into the stripe API a bit but I'm guessing you could use a redirect URI to tell the FE to post it as a SuperChat once the payment goes through

**Comment by dm17** [comment]
OK, sounds like you're thinking more low level. I think whatever chat software is integrated would need some added features to support that too. The main reason for sending money is for your chat/question to stand out. Many people do live streaming these days for Q/A and interaction with their audience to support them financially. Like an expert answering tech questions live, for example.

**Comment by GRVYDEV** [comment]
> OK, sounds like you're thinking more low level. I think whatever chat software is integrated would need some added features to support that too. The main reason for sending money is for your chat/question to stand out. Many people do live streaming these days for Q/A and interaction with their audience to support them financially. Like an expert answering tech questions live, for example.

Yeah I totally understand that and my original plan was to build our own chat API since I want to keep the number of outside code bases down and also a lot of realtime APIs cost money at scale.

**Comment by dm17** [comment]
Great to hear... Definitely post back here if you're having Stripe API-related issues; I might be able to help.

I also wonder if an existing OBS plugin (like streamelements.com/obslive - pretty sure it is OSS) could be extended to work with payments...
